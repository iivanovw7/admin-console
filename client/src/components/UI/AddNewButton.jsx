import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import withWidth from '@material-ui/core/withWidth';

const AddNewButton = props => {

  const { classes, width, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  const mobileButton = () => {
    return (
      <Button size="small" className={classes.margin}
              onClick={() => {
                handleClick();
              }}>
        NEW
      </Button>
    );
  };

  const desktopButton = () => {
    return (
      <Fab color="primary" size="small" aria-label="Add" className={classes.fab}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    );
  };

  const components = {
    xs: mobileButton(),
    sm: mobileButton()
  };

  const Component = components[width] || desktopButton();

  return (
    <div className={classes.formRoot}>
      {Component}
    </div>
  );

};

AddNewButton.propTypes = {
  width: PropTypes.string.isRequired
};

export default withWidth()(AddNewButton);

