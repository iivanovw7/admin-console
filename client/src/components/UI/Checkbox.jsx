import Checkbox from '@material-ui/core/Checkbox';
import PropTypes from 'prop-types';
import React from 'react';

export const ActionCheckBox = props => {

  const { status, classes, handleClick } = props;

  return (
    <Checkbox
      className={classes.actionCheckBox}
      checked={status}
      size={'small'}
      color={'primary'}
      onChange={handleClick}
      disabled={false}
    />
  );

};

ActionCheckBox.propTypes = {
  classes: PropTypes.object,
  handleClick: PropTypes.func,
  status: PropTypes.bool
};
