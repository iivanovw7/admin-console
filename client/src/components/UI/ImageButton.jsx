import { Button, IconButton, ListItem } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';

export const ImageButton = props => {

  const { classes, image, alt, handleClick, title } = props;

  return (
    <ListItem style={{ padding: 0, marginTop: 3, marginBottom: 3 }}>
      <IconButton className={classes.desktop} color="default" aria-label={alt} onClick={handleClick}
                  size={'small'}>
        <img style={{ width: '24px' }} src={image} alt={alt}/>
      </IconButton>
      <Button
        style={{ fontSize: '0.5rem' }}
        className={classes.mobile} size="small"
        onClick={handleClick}
      >
        {title}
      </Button>
    </ListItem>
  );
};

ImageButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  image: PropTypes.any.isRequired
};

