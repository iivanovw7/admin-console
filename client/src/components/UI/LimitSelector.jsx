import React from 'react';
import InputLabel from '@material-ui/core/InputLabel';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

export const LimitSelector = props => {

  const { classes, limit, handleLimit, limits } = props;

  return (
    <div className={classes.fromLimitSelector}>
      <FormControl className={classes.formControl}>
        <InputLabel>Results limit</InputLabel>
        <Select
          native
          value={limit}
          onChange={handleLimit}
        >
          {limits.map(newLimit => (
            <option key={newLimit} value={newLimit}>
              {newLimit}
            </option>
          ))}
        </Select>
      </FormControl>
    </div>
  );

};

