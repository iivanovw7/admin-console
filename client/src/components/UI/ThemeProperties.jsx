import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

