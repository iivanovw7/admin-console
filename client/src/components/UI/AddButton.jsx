import { withStyles } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Button from '@material-ui/core/Button';
import { Buttons } from './ThemeProperties';

const AddButton = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button size="small" className={classes.mobile}
              onClick={() => {
                handleClick();
              }}>
        <strong>Create</strong>
      </Button>
      <Fab color="primary" size="small" aria-label="Add" className={classes.desktop}
           onClick={() => {
             handleClick();
           }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

AddButton.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(AddButton);