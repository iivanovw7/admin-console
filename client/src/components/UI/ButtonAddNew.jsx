import { Button, Fab } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import PropTypes from 'prop-types';
import React from 'react';
import { Buttons } from './ThemeProperties';

const ButtonAddNew = props => {

  const { classes, history, element } = props;

  function handleClick() {
    return history.push(history.push(`/${element}/new`));
  }

  return (
    <div className={classes.formRoot}>
      <Button
        size="small"
        className={classes.mobile}
        onClick={() => {
          handleClick();
        }}
      >
        <strong>Create</strong>
      </Button>
      <Fab
        color="primary"
        size="small"
        aria-label="Add"
        className={classes.desktop}
        onClick={() => {
          handleClick();
        }}>
        <AddIcon/>
      </Fab>
    </div>
  );

};

ButtonAddNew.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(Buttons, { withTheme: true })(ButtonAddNew);