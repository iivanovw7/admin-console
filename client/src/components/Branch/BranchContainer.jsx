import { Button, Grid } from '@material-ui/core';
import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { reduxForm } from 'redux-form';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import { TextInputContainer, CheckboxContainer } from './Form/InputContainers';
import { errorMessage } from './Form/ErrorMessage';
import { validate } from './Form/Validation';
import { Branches } from '../UI/ThemeProperties';
import { addBranch, updateBranch } from '../../actions/branches';
import normalizePhone from './Form/Normalizer';

const BranchContainer = props => {

  const { classes, history, handleSubmit } = props;
  const branch = props.branches.branch;

  useEffect(() => {
    props.initialize({
      name: branch.name || '',
      email: branch.email || '',
      phone: branch.phone || '',
      fax: branch.fax || '',
      address: branch.address || '',
      information: branch.information || '',
      status: branch.status || false
    });
  }, [branch]);

  const submit = (formValues) => {
    return (!branch._id) ?
      (props.addBranch(formValues, history)) : (props.updateBranch(formValues, branch._id, history));
  };

  return (
    <Paper className={classes.root}>
      <form className={classes.branchPaper} onSubmit={handleSubmit(submit)}>
        <br/>
        <TextInputContainer dataType={'name'} type={'text'} rows={1}/>
        <TextInputContainer dataType={'email'} type={'email'} rows={1}/>
        <TextInputContainer dataType={'phone'} type={'phone'} rows={1} normalize={normalizePhone}/>
        <TextInputContainer dataType={'fax'} type={'fax'} rows={1} normalize={normalizePhone}/>
        <TextInputContainer dataType={'address'} type={'text'} rows={4}/>
        <TextInputContainer dataType={'information'} type={'text'} rows={4}/>
        <CheckboxContainer name={'status'} label={'Active'} value={''}/>
        <Grid container justify="flex-end" style={{ marginTop: '10px' }}>
          <Button
            variant="outlined" color="primary" style={{ textTransform: 'none', margin: 5 }}
            onClick={() => {
              history.push(history.push(`/branches`));
            }}>
            CANCEL
          </Button>
          <Button variant="outlined" color="primary" type="submit"
                  style={{ textTransform: 'none', margin: 5 }}>SAVE</Button>
          {errorMessage(props)}
        </Grid>
      </form>
    </Paper>
  );
};

BranchContainer.propTypes = {
  classes: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func
};

function mapStateToProps(state) {
  return {
    errorMessage: state.branches.error,
    branches: state.branches
  };
}

const reduxFromBranch = reduxForm({
  validate,
  form: 'branch',
  fields: ['name', 'email', 'phone', 'fax', 'address', 'information', 'status']
})(BranchContainer);

export default connect(mapStateToProps, {
  addBranch,
  updateBranch
})(withStyles(Branches)(reduxFromBranch));
