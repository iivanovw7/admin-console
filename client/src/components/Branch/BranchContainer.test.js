import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import jest from 'jest-mock';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { reduxForm } from 'redux-form';
import { addNewBranch, updateBranch } from '../../actions';
import BranchContainer from '../../components/Branch/BranchContainer';
import { validateBranch } from '../../utils/formsValidator';
import { Container } from '../../components/UI/ThemeProperties';
import { storeFactory, history } from '../../testUtils';
import * as mocks from '../../__mocks__/';

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);

  //Splitting HOC components
  const Form = reduxForm({
    validate: validateBranch,
    asyncBlurFields: [], //to avoid error related to async validation
    form: 'branch',
    fields: [
      'name',
      'email',
      'phone',
      'fax',
      'address',
      'information',
      'status']
  })(BranchContainer);

  const Composition = connect(null, {
    addNewBranch,
    updateBranch
  })(withStyles(Container, { withTheme: true })(Form));

  const props = {
    branch: mocks.BRANCH,
    history: history
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders branch form with styles, imitates users actions:
   *  clicks buttons and input fields
   */
  describe('Branch Container', () => {
    it('Should render branch form with styles', () => {
      const mockFunc = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <Composition {...props} onSubmit={mockFunc()} />
        </Provider>
      );

      expect(wrapper.html()).toMatchSnapshot();
      expect(wrapper.find('button[type="button"]').text()).toBe('CANCEL');
      expect(wrapper.find('button[type="submit"]').text()).toBe('SAVE');

      //Comparing rendered fields to mock data
      expect(wrapper.find(BranchContainer).at(0).props(0).branch.address).toBe(mocks.BRANCH.address);
      expect(wrapper.find(BranchContainer).at(0).props(0).branch).toBe(mocks.BRANCH);

      //Checking email input field
      wrapper.find('input#email').simulate('click');
      wrapper.find('input#email').simulate('change', {target: { value: 'email@test.com'}});
      expect(wrapper.find('input#email').props().value).toBe('email@test.com');

      //Checking address input field
      wrapper.find('input#name').simulate('click');
      wrapper.find('input#name').simulate('change', {target: { value: 'Test name'}});
      expect(wrapper.find('input#name').props().value).toBe('Test name');

      //checking submit event handling
      wrapper.find(BranchContainer).simulate('submit');
      expect(mockFunc).toHaveBeenCalledTimes(1);

    });
  });
};

setup();
