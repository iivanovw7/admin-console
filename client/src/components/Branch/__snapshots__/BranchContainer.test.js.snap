// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branch Container Should render branch form with styles 1`] = `"<div class=\\"MuiPaper-root-59 MuiPaper-elevation2-63 MuiPaper-rounded-60 ReduxForm-root-30 ReduxForm-root-1\\"><form class=\\"ReduxForm-branchPaper-44 ReduxForm-branchPaper-15\\"><br><div class=\\"MuiGrid-container-86 MuiGrid-spacing-xs-16-109 MuiGrid-align-items-xs-flex-end-96\\"><div class=\\"MuiGrid-item-87 MuiGrid-grid-xs-true-114 MuiGrid-grid-sm-true-128 MuiGrid-grid-md-true-142\\"><div class=\\"MuiFormControl-root-183 MuiFormControl-fullWidth-186\\"><label class=\\"MuiFormLabel-root-198 MuiFormLabel-filled-202 MuiInputLabel-root-187 MuiInputLabel-formControl-192 MuiInputLabel-animated-195 MuiInputLabel-shrink-194 MuiInputLabel-outlined-197\\" data-shrink=\\"true\\" for=\\"name\\">name</label><div class=\\"MuiInputBase-root-218 MuiOutlinedInput-root-205 MuiInputBase-fullWidth-227 MuiInputBase-formControl-219\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-235 MuiOutlinedInput-notchedOutline-212\\"><legend class=\\"MuiPrivateNotchedOutline-legend-236\\" style=\\"width: 0px;\\"><span>​</span></legend></fieldset><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-228 MuiOutlinedInput-input-213 MuiInputBase-inputType-231\\" id=\\"name\\" name=\\"name\\" rows=\\"1\\" type=\\"name\\" value=\\"San Martín de los Andes\\"></div></div></div></div><div class=\\"MuiGrid-container-86 MuiGrid-spacing-xs-16-109 MuiGrid-align-items-xs-flex-end-96\\"><div class=\\"MuiGrid-item-87 MuiGrid-grid-xs-true-114 MuiGrid-grid-sm-true-128 MuiGrid-grid-md-true-142\\"><div class=\\"MuiFormControl-root-183 MuiFormControl-fullWidth-186\\"><label class=\\"MuiFormLabel-root-198 MuiFormLabel-filled-202 MuiInputLabel-root-187 MuiInputLabel-formControl-192 MuiInputLabel-animated-195 MuiInputLabel-shrink-194 MuiInputLabel-outlined-197\\" data-shrink=\\"true\\" for=\\"email\\">email</label><div class=\\"MuiInputBase-root-218 MuiOutlinedInput-root-205 MuiInputBase-fullWidth-227 MuiInputBase-formControl-219\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-235 MuiOutlinedInput-notchedOutline-212\\"><legend class=\\"MuiPrivateNotchedOutline-legend-236\\" style=\\"width: 0px;\\"><span>​</span></legend></fieldset><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-228 MuiOutlinedInput-input-213 MuiInputBase-inputType-231\\" id=\\"email\\" name=\\"email\\" rows=\\"1\\" type=\\"email\\" value=\\"mbrazier0@amazon.co.uk\\"></div></div></div></div><div class=\\"MuiGrid-container-86 MuiGrid-spacing-xs-16-109 MuiGrid-align-items-xs-flex-end-96\\"><div class=\\"MuiGrid-item-87 MuiGrid-grid-xs-true-114 MuiGrid-grid-sm-true-128 MuiGrid-grid-md-true-142\\"><div class=\\"MuiFormControl-root-183 MuiFormControl-fullWidth-186\\"><label class=\\"MuiFormLabel-root-198 MuiFormLabel-filled-202 MuiInputLabel-root-187 MuiInputLabel-formControl-192 MuiInputLabel-animated-195 MuiInputLabel-shrink-194 MuiInputLabel-outlined-197\\" data-shrink=\\"true\\" for=\\"phone\\">phone</label><div class=\\"MuiInputBase-root-218 MuiOutlinedInput-root-205 MuiInputBase-fullWidth-227 MuiInputBase-formControl-219\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-235 MuiOutlinedInput-notchedOutline-212\\"><legend class=\\"MuiPrivateNotchedOutline-legend-236\\" style=\\"width: 0px;\\"><span>​</span></legend></fieldset><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-228 MuiOutlinedInput-input-213 MuiInputBase-inputType-231\\" id=\\"phone\\" name=\\"phone\\" rows=\\"1\\" type=\\"phone\\" value=\\"+54 275 474 9318\\"></div></div></div></div><div class=\\"MuiGrid-container-86 MuiGrid-spacing-xs-16-109 MuiGrid-align-items-xs-flex-end-96\\"><div class=\\"MuiGrid-item-87 MuiGrid-grid-xs-true-114 MuiGrid-grid-sm-true-128 MuiGrid-grid-md-true-142\\"><div class=\\"MuiFormControl-root-183 MuiFormControl-fullWidth-186\\"><label class=\\"MuiFormLabel-root-198 MuiFormLabel-filled-202 MuiInputLabel-root-187 MuiInputLabel-formControl-192 MuiInputLabel-animated-195 MuiInputLabel-shrink-194 MuiInputLabel-outlined-197\\" data-shrink=\\"true\\" for=\\"fax\\">fax</label><div class=\\"MuiInputBase-root-218 MuiOutlinedInput-root-205 MuiInputBase-fullWidth-227 MuiInputBase-formControl-219\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-235 MuiOutlinedInput-notchedOutline-212\\"><legend class=\\"MuiPrivateNotchedOutline-legend-236\\" style=\\"width: 0px;\\"><span>​</span></legend></fieldset><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input-228 MuiOutlinedInput-input-213 MuiInputBase-inputType-231\\" id=\\"fax\\" name=\\"fax\\" rows=\\"1\\" type=\\"fax\\" value=\\"+86 829 425 7497\\"></div></div></div></div><div class=\\"MuiGrid-container-86 MuiGrid-spacing-xs-16-109 MuiGrid-align-items-xs-flex-end-96\\"><div class=\\"MuiGrid-item-87 MuiGrid-grid-xs-true-114 MuiGrid-grid-sm-true-128 MuiGrid-grid-md-true-142\\"><div class=\\"MuiFormControl-root-183 MuiFormControl-fullWidth-186\\"><label class=\\"MuiFormLabel-root-198 MuiFormLabel-filled-202 MuiInputLabel-root-187 MuiInputLabel-formControl-192 MuiInputLabel-animated-195 MuiInputLabel-shrink-194 MuiInputLabel-outlined-197\\" data-shrink=\\"true\\" for=\\"address\\">address</label><div class=\\"MuiInputBase-root-218 MuiOutlinedInput-root-205 MuiInputBase-fullWidth-227 MuiInputBase-formControl-219 MuiInputBase-multiline-226 MuiOutlinedInput-multiline-211\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-235 MuiOutlinedInput-notchedOutline-212\\"><legend class=\\"MuiPrivateNotchedOutline-legend-236\\" style=\\"width: 0px;\\"><span>​</span></legend></fieldset><div class=\\"MuiPrivateTextarea-root-237\\"><textarea aria-hidden=\\"true\\" class=\\"MuiPrivateTextarea-textarea-238 MuiPrivateTextarea-shadow-239\\" readonly=\\"\\" rows=\\"1\\" tabindex=\\"-1\\"></textarea><textarea aria-hidden=\\"true\\" class=\\"MuiPrivateTextarea-textarea-238 MuiPrivateTextarea-shadow-239\\" readonly=\\"\\" rows=\\"4\\" tabindex=\\"-1\\">938 Barby Way</textarea><textarea rows=\\"4\\" class=\\"MuiPrivateTextarea-textarea-238 MuiInputBase-input-228 MuiOutlinedInput-input-213 MuiInputBase-inputType-231 MuiInputBase-inputMultiline-230 MuiOutlinedInput-inputMultiline-215\\" style=\\"height: 19px;\\" aria-invalid=\\"false\\" id=\\"address\\" name=\\"address\\">938 Barby Way</textarea></div></div></div></div></div><div class=\\"MuiGrid-container-86 MuiGrid-spacing-xs-16-109 MuiGrid-align-items-xs-flex-end-96\\"><div class=\\"MuiGrid-item-87 MuiGrid-grid-xs-true-114 MuiGrid-grid-sm-true-128 MuiGrid-grid-md-true-142\\"><div class=\\"MuiFormControl-root-183 MuiFormControl-fullWidth-186\\"><label class=\\"MuiFormLabel-root-198 MuiFormLabel-filled-202 MuiInputLabel-root-187 MuiInputLabel-formControl-192 MuiInputLabel-animated-195 MuiInputLabel-shrink-194 MuiInputLabel-outlined-197\\" data-shrink=\\"true\\" for=\\"information\\">information</label><div class=\\"MuiInputBase-root-218 MuiOutlinedInput-root-205 MuiInputBase-fullWidth-227 MuiInputBase-formControl-219 MuiInputBase-multiline-226 MuiOutlinedInput-multiline-211\\"><fieldset aria-hidden=\\"true\\" style=\\"padding-left: 8px;\\" class=\\"MuiPrivateNotchedOutline-root-235 MuiOutlinedInput-notchedOutline-212\\"><legend class=\\"MuiPrivateNotchedOutline-legend-236\\" style=\\"width: 0px;\\"><span>​</span></legend></fieldset><div class=\\"MuiPrivateTextarea-root-237\\"><textarea aria-hidden=\\"true\\" class=\\"MuiPrivateTextarea-textarea-238 MuiPrivateTextarea-shadow-239\\" readonly=\\"\\" rows=\\"1\\" tabindex=\\"-1\\"></textarea><textarea aria-hidden=\\"true\\" class=\\"MuiPrivateTextarea-textarea-238 MuiPrivateTextarea-shadow-239\\" readonly=\\"\\" rows=\\"4\\" tabindex=\\"-1\\">1</textarea><textarea rows=\\"4\\" class=\\"MuiPrivateTextarea-textarea-238 MuiInputBase-input-228 MuiOutlinedInput-input-213 MuiInputBase-inputType-231 MuiInputBase-inputMultiline-230 MuiOutlinedInput-inputMultiline-215\\" style=\\"height: 19px;\\" aria-invalid=\\"false\\" id=\\"information\\" name=\\"information\\">1</textarea></div></div></div></div></div><div class=\\"MuiGrid-container-86 MuiGrid-spacing-xs-16-109 MuiGrid-align-items-xs-flex-end-96\\"><div class=\\"MuiGrid-item-87 MuiGrid-grid-xs-true-114 MuiGrid-grid-sm-true-128 MuiGrid-grid-md-true-142\\"><label class=\\"MuiFormControlLabel-root-240\\"><span class=\\"MuiButtonBase-root-262 MuiIconButton-root-256 MuiPrivateSwitchBase-root-252 MuiCheckbox-root-246 MuiCheckbox-colorPrimary-250 MuiPrivateSwitchBase-checked-253 MuiCheckbox-checked-247\\"><span class=\\"MuiIconButton-label-261\\"><svg class=\\"MuiSvgIcon-root-265\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"></path></svg><input class=\\"MuiPrivateSwitchBase-input-255\\" id=\\"status\\" name=\\"status\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\"></span><span class=\\"MuiTouchRipple-root-336\\"></span></span><span class=\\"MuiTypography-root-274 MuiTypography-body1-283 MuiFormControlLabel-label-245\\">Active</span></label></div></div><div class=\\"MuiGrid-container-86 MuiGrid-justify-xs-flex-end-104\\" style=\\"margin-top: 10px;\\"><button class=\\"MuiButtonBase-root-262 MuiButton-root-310 MuiButton-contained-321 MuiButton-containedPrimary-322 MuiButton-raised-324 MuiButton-raisedPrimary-325\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform: none; margin: 5px;\\"><span class=\\"MuiButton-label-311\\">CANCEL</span><span class=\\"MuiTouchRipple-root-336\\"></span></button><button class=\\"MuiButtonBase-root-262 MuiButton-root-310 MuiButton-contained-321 MuiButton-containedPrimary-322 MuiButton-raised-324 MuiButton-raisedPrimary-325\\" tabindex=\\"0\\" type=\\"submit\\" style=\\"text-transform: none; margin: 5px;\\"><span class=\\"MuiButton-label-311\\">SAVE</span><span class=\\"MuiTouchRipple-root-336\\"></span></button></div></form></div>"`;
